### DISCUSSION TITLE
bug in GH UI? shows code scanning alerts as 1 but there are several thousand

### DISCUSSION THREAD

**Discussion by ngamboa** [community/community#102058] [Code Security]
### Select Topic Area

Bug

### Body

If you look at the image below, on the left nav bar there is code scanning = 1, however on the right results pane there are over 7k. This looks like a UI bug, i have refreshed multiple time in case it was a latency glitch.
![image](https://github.com/community/community/assets/4173761/9e4adbd2-a68e-4f49-971a-a4c738ba73d1)

**Comment by SujalMeghwal** [comment]
What's Going On
Left Sidebar Bug: The left navigation bar in the GitHub UI shows Code scanning = 1.

Main Panel Reality: The main results pane shows over 7,000 open code scanning alerts.

User Observation: You mentioned refreshing didn’t resolve it, suggesting it’s not a simple cache or latency issue.

Root Cause Possibilities
Frontend UI Desync / Caching Bug:

GitHub UI sometimes caches sidebar counts aggressively.

The count in the sidebar may not be refreshing in real-time or is stuck due to a caching issue or partial fetch.

Permission/Scope Filtering:

If you’re not the repo owner or lack full access, certain alerts may be hidden in the sidebar count.

The UI may be showing "1 visible" based on your permission, but the actual number of alerts is higher in the main pane due to filters or backend inconsistencies.

API Indexing Delay or Bug:

GitHub’s backend may not have re-indexed the code scanning alerts properly.

The sidebar pulls from a different API endpoint or cache layer than the detailed view.

Multi-Repo View / Aggregated Bugs:

If you're looking at an organization security overview (vs. per-repo), sometimes the aggregation misbehaves.

GitHub may be showing an incorrect summary if some repos are excluded from the count.

🛠️ How to Troubleshoot and Solve
✅ Immediate Steps You Can Try
Clear Cache and Cookies:

Log out and log in again.

Try incognito mode or a different browser.

Use the GitHub API Directly:

Run the following GitHub API call to verify how many code scanning alerts actually exist:

curl -H "Authorization: token <YOUR_TOKEN>" https://api.github.com/repos/<owner>/<repo>/code-scanning/alerts
This will give you the real backend count and help confirm it’s a UI bug.

Check Repository Permissions:

If you're not the admin of the repo, ask the repo owner to confirm alert visibility.

GitHub hides some alerts from users without specific security privileges.

Try Forcing a Reindex:

Push a dummy commit to the main branch.

This sometimes triggers GitHub to re-scan and refresh UI counts.

🧪 Deeper Debugging (If You Control the Org/Repo)
Go to Settings > Code Security and Analysis and make sure CodeQL scanning is enabled for all branches.

Re-run the analysis manually:

gh codeql analyze --repository=<repo>
